### ISSUE TITLE
[profiler] Perform simple tests and plot the results with different configurations

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1120]
cc'ing @pablogoitia 

This issue is a follow-up to #1051 and #1108.

Some suggestions of test scenarios:

1. A workflow configuration similar to what Dani used to profile memory issues
2. A workflow configuration with no placeholders, compared to a workflow configuration with lots of placeholders
3. A workflow configuration with minimal, compared to a workflow configuration with old-style configuration
4. A workflow configuration with minimal, compared to a workflow configuration with minimal and many CUSTOM_CONFIG's, PRE's, POST's
5. A workflow configuration with a single minimal file, that contains a lot of extra configuration entries
6. A workflow configuration with a collection of many (100's, 1000's) of small YAML files.
7. A workflow configuration with a collection of many (100's, 1000's) of large YAML files.
8. A workflow configuration with minimal, that includes another YAML file
9. A workflow configuration with wrappers
  9.1. Horizontal
  9.2. Vertical
  9.3. Hybrid
...

We can create a directory in the code, maybe under test (e.g. `test/performance/profiler-tests`).

Each test could be a Jupyter notebook (bonus points if using Docker). It must contain a short paragraph containing the analysis of the results, followed by a simple 2-bar chart (matplotlib/notebook?) showing memory consumption side-by-side. We can also filter the top-N entries of the cProfile output (probably `N=5` is good enough?).

The tests can be performed locally or on our VM's or MN4. I think it won't make much difference running on MN4, since the Python version and libraries can be replicated locally. IMHO, the only advantage of using MN4 would be if we were measuring I/O.

**Comment by kinow** [comment]
@pablogoitia I don't believe we will have time to finish it this week, but I will try to do at least one or two and add a draft merge request so that you can have a look :thumbsup:

**Comment by kinow** [comment]
In GitLab by @pablogoitia on Aug 22, 2023, 10:45

Nice! Thanks @kinow :blush:

**Comment by kinow** [comment]
> A workflow configuration similar to what Dani used to profile memory issues

![image](https://earth.bsc.es/gitlab/es/autosubmit/uploads/74894a94d979654078dbee28eb474f16/image.png)

**Comment by dbeltrankyl** [comment]
Very old issue, fyi @kinow as you have an open branch for this
